docs/SQL.md

```diff
@@ -0,0 +1,122 @@
+
+###### 6.Avoiding Data Redundancy:
+Instead of duplicating data across multiple tables, which can lead to redundancy and inconsistency, you can store data in separate tables and use joins to retrieve consolidated information as needed. This approach adheres to the principles of database normalization.
+
+###### 7.Aggregating Data:
+Joins are often used in combination with aggregate functions (e.g., COUNT, SUM, AVG) to perform calculations and analysis on data from multiple tables.
+
+###### 8.Enhancing Reporting and Analysis:
+For reporting and business intelligence purposes, joins are essential to gather and present comprehensive data that may be distributed across various tables.
+
+###### 9.Complex Queries:
+Complex queries involving multiple conditions and criteria often require joins to bring together the necessary data from different tables.
+
+It's important to choose the appropriate type of join based on the desired outcome. The most common join types include INNER JOIN, LEFT JOIN (or LEFT OUTER JOIN), RIGHT JOIN (or RIGHT OUTER JOIN), and FULL JOIN (or FULL OUTER JOIN), each serving specific purposes in combining data from related tables.
+
+
+## Explain 7 types of Joins
+
+#### 1.INNER JOIN:
+An INNER JOIN returns rows from both tables where there is a match based on the specified condition. Rows from tables that do not meet the condition are excluded from the result set.
+
+###### Example:
+SELECT * FROM table1 INNER JOIN table2 ON table1.column = table2.column;
+
+#### 2.LEFT JOIN (or LEFT OUTER JOIN):
+
+A LEFT JOIN returns all rows from the left table and the matching rows from the right table. If there is no match, NULL values are returned for columns from the right table.
+
+###### Example:
+SELECT * FROM table1 LEFT JOIN table2 ON table1.column = table2.column;
+
+#### 3.RIGHT JOIN (or RIGHT OUTER JOIN):
+
+A RIGHT JOIN returns all rows from the right table and the matching rows from the left table. If there is no match, NULL values are returned for columns from the left table.
+
+###### Example:
+SELECT * FROM table1 RIGHT JOIN table2 ON table1.column = table2.column;
+
+#### 4.FULL JOIN (or FULL OUTER JOIN):
+
+A FULL JOIN returns all rows when there is a match in either the left or right table. If there is no match, NULL values are returned for columns from the table without a match.
+
+###### Example:
+SELECT * FROM table1 FULL JOIN table2 ON table1.column = table2.column;
+
+#### 5.CROSS JOIN:
+
+A CROSS JOIN returns the Cartesian product of the two tables, i.e., all possible combinations of rows from both tables. It doesn't use any join condition.
+
+###### Example:
+SELECT * FROM table1 CROSS JOIN table2;
+
+#### 6. SELF JOIN:
+
+A SELF JOIN is used to join a table with itself. This is often done when comparing rows within the same table.
+
+###### Example:
+SELECT * FROM table1 t1 INNER JOIN table1 t2 ON t1.column = t2.column;
+
+#### 7. ANTI JOIN (or ANTI SEMI-JOIN):
```

Comment:
The description explains the LEFT ANTI JOIN but heading is mentioned as ANTI JOIN.

The Performance of the Query can be Optimized by applying FULL OUTER JOIN and selecting only those rows where the right part is NULL.

Reply:
Yeah , my mistake...
So , i need to change the description.